design/render.go

```diff
@@ -5,34 +5,64 @@ import (
 	a "github.com/goadesign/goa/design/apidsl"
 )
 
-// MarkupRenderingInputType is the media type representing a rendering input.
-var MarkupRenderingInputType = a.Type("MarkupRenderingInputType", func() {
+// MarkupRenderingPayloadData is the media type representing a rendering input.
+var markupRenderingPayloadData = a.Type("MarkupRenderingPayloadData", func() {
 	a.Description("A render type describes the values a render type instance can hold.")
-	a.Attribute("markup", d.String, "The name of the markup language used to input the content")
-	a.Attribute("content", d.String, "The content to be rendered")
-	a.Required("markup")
-	a.Required("content")
+	a.Attribute("id", d.String, "an ID to conform to the JSON-API spec, even though it is meaningless in the case of the rendering endpoint. Can be null", func() {
+		a.Example("42")
+	})
+	a.Attribute("type", d.String, func() {
+		a.Enum("rendering")
+	})
+	a.Attribute("attributes", a.HashOf(d.String, d.Any), func() {
```

Comment:
I'd suggest to make this a distinct Type instead of a simple hash. You do required the markup and content members, don't you?